### PR TITLE
define topic introspection API

### PR DESCRIPTION
**Pull request by dirk-thomas**
Connects to ros2/ros1_bridge#3

### CODE REVIEW

File: rmw/include/rmw/rmw.h [hunk 1]
```diff
@@ -152,6 +152,32 @@ rmw_wait(
   rmw_clients_t * clients,
   bool non_blocking);
 
+RMW_PUBLIC
+rmw_ret_t
+rmw_get_topics(
+  const rmw_node_t * node,
+  rmw_topic_names_t * topic_names);
```
Comment (wjwwood) [inline rmw/include/rmw/rmw.h]: You'll probably want a matching `destroy_topic_names` function to undo any allocation done by this function.

**Comment by wjwwood** [comment]
+1 other than comments.

**Comment by dirk-thomas** [comment]
Ready for review.

**Comment by esteve** [comment]
+1

**Comment by jacquelinekay** [comment]
Waiting on rebase and CI for the `topic_information` branches before I approve.

**Pull request merged**